mbed_host_tests/host_tests_conn_proxy/conn_proxy.py

```diff
@@ -174,11 +174,11 @@ def conn_process(event_queue, dut_event_queue, prn_lock, config):
             break
 
         # Send data to DUT
-        if not dut_event_queue.empty():
-            try:
-                (key, value, _) = dut_event_queue.get(timeout=1)
-            except QueueEmpty:
-                continue
+        try:
+            (key, value, _) = dut_event_queue.get(timeout=0)
```

Comment:
[method Queue.get(block, timeout)](https://docs.python.org/2/library/queue.html#Queue.Queue.get)
> Remove and return an item from the queue. If optional args block is true and timeout is None (the default), block if necessary until an item is available. **If timeout is a positive number, it blocks at most timeout seconds and raises the Empty exception if no item was available within that time**. Otherwise (block is false), return an item if one is immediately available, else raise the Empty exception (timeout is ignored in that case).

Are you sure setting timeout to 0 will raise `Empty` exception (`QueueEmpty` is alias for `Empty`).